tools/flow9/backends/cpp/cpp.flow

```diff
@@ -160,49 +161,84 @@ bstatement2cpp(backend : Backend<CppInfo>, s : BStatement) -> string {
 				""
 			} else {
 				"return ";
-			}) + bexp2cpp(backend, val) + ";"
+			}) + bexp2cpp(backend, val, type, "") + ";"
 		}
 		BTailCall(fn, args, values, type): {
-			fn  + "(" +superglue(values, rece, ", ") + ");"
+			fn  + "(" +superglue(values, \v -> rece(v, type), ", ") + ");"
```

Comment:
This is wrong, if it is expected type that is required. You have to get the type of each argument in the function type from the fn.